orchestrator/src/server.ts

```diff
@@ -167,23 +167,43 @@ async function runPreprocessorsParallel(data: Record<string, unknown>, preproces
         data["preprocessors"] = {};
     }
     let currentPriorityGroup: number | undefined = undefined;
-    let promises: Promise<void>[] = []; // array to hold promises for preprocessor executions within the current priority group
+    const queue: (string | number)[][] = []; //Microservice queue for preprocessors and handlers
+
+    const processQueue = async (): Promise<void> => {
+        const promises: Promise<void>[] = [];   //store promises of each preprocessor 
+
+        while (queue.length > 0) {
+            const preprocessor = queue.shift() as (string | number)[];  //dequeue preprocessor 
+            if (preprocessor) { //ensure preprocessor is defined
+                //Wait for the preprocessor to complete execution before moving to the next
+                promises.push(executePreprocessor(preprocessor, data));
+            }
+        }
+        try {
+            await Promise.all(promises);    //wait for all promises in the current priority group to finish
+        } catch (error) {
+            console.error(`One or more of the promises failed at ${currentPriorityGroup} priority group.`, error);
+        }
+          
```

Comment:
Could you add a comment to this noting that this function runs all contents of the queue concurrently and waits for them to finish?